compiler/rustc_target/src/spec/wasm32_unknown_emscripten.rs

```diff
@@ -27,6 +27,7 @@ pub fn target() -> Target {
         exe_suffix: ".js".into(),
         linker: None,
         is_like_emscripten: true,
+        relocation_model: RelocModel::Pic,
```

Comment:
Does this mean that all rust emscripten projects will always be PIC? 

Its not a huge deal since the linker can deal with PIC object when linking static/non-PIC binaries, but there could be some overhead in the object files and in the un-optimized-by-binaryen wasm output.

Reply:
I think that's correct. It seems that a target can either have everything PIC or nothing PIC.

Reply:
I guess the alternative would be to have two separate compiler targets, one for PIC and one for static. But none of the other targets with PIC seem to be set up like this.

Reply:
Sounds reasonable.  In older versions of emscripten PIC objects was not compatible with non-PIC binaries, but today it should be (with the caveats above that I'm fairly sure doesn't effect optimized builds).

Reply:
>It seems that a target can either have everything PIC or nothing PIC.

What do you mean by that?
There's an option [`-Crelocation_model`](https://doc.rust-lang.org/rustc/codegen-options/index.html#relocation-model) to override the target default for a specific binary.